src/gordon_janitor_gcp/clients/gce.py

```diff
@@ -145,6 +145,7 @@ def _extract_instance_data(self, instance):
             'hostname': instance['name'],
             'internal_ip': iface_data['networkIP'],
             'external_ip': iface_data['accessConfigs'][0]['natIP'],
+            'creation_timestamp': instance['creationTimestamp']
```

Comment:
I'm confused at why this is needed here, and not our internal janitor gcp plugin that would inherit from this? I probably just can't think of a use case for others.

Reply:
It isn't *needed* and I considered the option of not including it here (instead subclassing the gceclient and using that internal gceclient in our internal subclassed plugin), but it didn't seem unreasonable to me that another plugin might like having access to this data as well, for example if a janitor only wanted to consider instances of a certain age, or use it somehow to deterministically set ttl.  

I also considered dropping this method altogether, because it seems strange that the default gce client should be enforcing opinions of what data is relevant rather than just passing back the raw instance data.  In the end, I decided to keep it to lower the memory consumption of the list of instances but since we're not sure there will be other consumers of this class, I altered it that it should at least cover our use case, since all the others are hypothetical.  I'm not married to it, though.

Reply:
Another option I didn't consider until now, could be to make it configurable somehow, via the toml, like `gcp.gce.additional_instance_data_fields = {'creation_timestamp': 'creationTimestamp', 'metadata_item_XYZ': ['metadata', 'items', 'keyXYZ']}` (i.e new field name -> path to var in the instance data)?

Reply:
@lynn @matto any thoughts? I'm probably going to be picking up this PR again soon.  To clarify one of my comments above, the method that seems strange to me is `_extract_instance_data` and in this case "strange" to me is "why should the plugin that supplies the instance information be limiting what information is relevant?"

Reply:
This is a good point, it seems the class has more responsibility than it should and that maybe this extracting should happen in the authority class. This way, the `creationTimestamp` business could be limited to the internal plugin. @descent3me do you think this refactor could be a part of your current ticket?